js/src/lib/interfaces/Types.ts

```diff
@@ -1,64 +1,179 @@
+export interface SubTask {
+	type: string;
+	name: string;
+}
 
-/// Coarse-grained task classification.
-///
-/// This type is used to determine which inference API & widget
-/// we want to display for each given model.
-///
-/// As such, they describe the "shape" of each model's API (inputs and outputs)
-/// so the number of different types is not expected to grow very significantly over time.
-///
-/// In each category, order by decreasing specificity
-/// The order can influence which default pipeline tag is affected to a model (if unspecified in model card)
-export enum PipelineType {
-	/// nlp
-	"text-classification"                                     = "Text Classification",
-	"token-classification"                                    = "Token Classification",
-	"table-question-answering"                                = "Table Question Answering",
-	"question-answering"                                      = "Question Answering",
-	"zero-shot-classification"                                = "Zero-Shot Classification",
-	"translation"                                             = "Translation",
-	"summarization"                                           = "Summarization",
-	"conversational"                                          = "Conversational",
-	"feature-extraction"                                      = "Feature Extraction",
-	"text-generation"                                         = "Text Generation",
-	"text2text-generation"                                    = "Text2Text Generation",
-	"fill-mask"                                               = "Fill-Mask",
-	"sentence-similarity"                                     = "Sentence Similarity",
-	/// audio
-	"text-to-speech"                                          = "Text-to-Speech",
-	"automatic-speech-recognition"                            = "Automatic Speech Recognition",
-	"audio-to-audio"                                          = "Audio-to-Audio",
-	"audio-classification"                                    = "Audio Classification",
-	"voice-activity-detection"                                = "Voice Activity Detection",
-	/// computer vision
-	"image-classification"                                    = "Image Classification",
-	"object-detection"                                        = "Object Detection",
-	"image-segmentation"                                      = "Image Segmentation",
-	"text-to-image"                                           = "Text-to-Image",
-	"image-to-text"                                           = "Image-to-Text",
-	/// others
-	"structured-data-classification"                          = "Structured Data Classification",
-	"reinforcement-learning"                                  = "Reinforcement Learning",
+export interface PipelineData {
+	name: string;
+	subtasks: SubTask[];
```

Comment:
```suggestion
	subtasks?: SubTask[];
```
maybe make it optional to not have to specify an empty array below?